tfhe/src/high_level_api/integers/keys.rs

```diff
@@ -131,3 +131,36 @@ impl IntegerServerKey {
             .expect("Integer ServerKey is not initialized")
     }
 }
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct IntegerCompressedServerKey {
+    pub(crate) key: Option<crate::integer::CompressedServerKey>,
+    pub(crate) encryption_type: crate::shortint::EncryptionKeyChoice,
+}
+
+impl IntegerCompressedServerKey {
+    pub(in crate::high_level_api) fn new(client_key: &IntegerClientKey) -> Self {
+        let Some(integer_key) = &client_key.key else {
+            return Self {
+                key: None,
+                encryption_type: EncryptionKeyChoice::Big,
+            };
+        };
+        if client_key.wopbs_block_parameters.is_some() {
+            panic!("Cannot compress the ServerKey as it requires a function evaluation key");
+        }
```

Comment:
as it requires a function evaluation key, which is not currently compressible.

Otherwise the error message does not make much sense